pallets/liquid-staking/src/mock.rs

```diff
@@ -1,29 +1,225 @@
 };
 
-use xcm::v0::{Junction, MultiAsset, MultiLocation};
-
 use primitives::{Amount, Balance, CurrencyId, Rate, Ratio};
 
+use cumulus_primitives_core::ParaId;
+use orml_xcm_support::{IsNativeConcrete, MultiCurrencyAdapter};
+use pallet_xcm::XcmPassthrough;
+use polkadot_parachain::primitives::Sibling;
+pub use xcm::v0::{
+    Error as XcmError,
+    Junction::{self, GeneralKey, Parachain, Parent},
+    MultiAsset,
+    MultiLocation::{self, X1, X2, X3},
+    NetworkId, Xcm,
+};
+pub use xcm_builder::{
+    AccountId32Aliases, AllowTopLevelPaidExecutionFrom, AllowUnpaidExecutionFrom,
+    ChildParachainAsNative, ChildParachainConvertsVia, ChildSystemParachainAsSuperuser,
+    CurrencyAdapter as XcmCurrencyAdapter, EnsureXcmOrigin, FixedRateOfConcreteFungible,
+    FixedWeightBounds, IsConcrete, LocationInverter, NativeAsset, ParentAsSuperuser,
+    ParentIsDefault, RelayChainAsNative, SiblingParachainAsNative, SiblingParachainConvertsVia,
+    SignedAccountId32AsNative, SignedToAccountId32, SovereignSignedViaLocation, TakeWeightCredit,
+};
+use xcm_executor::{Config, XcmExecutor};
+use xcm_simulator::{decl_test_network, decl_test_parachain, decl_test_relay_chain};
+pub type AccountId = AccountId32;
+pub use westend_runtime;
+
+parameter_types! {
+    pub const ReservedXcmpWeight: Weight = WEIGHT_PER_SECOND / 4;
+    pub const ReservedDmpWeight: Weight = WEIGHT_PER_SECOND / 4;
+}
+
+impl cumulus_pallet_parachain_system::Config for Test {
+    type Event = Event;
+    type OnValidationData = ();
+    type SelfParaId = ParachainInfo;
+    type DmpMessageHandler = DmpQueue;
+    type ReservedDmpWeight = ReservedDmpWeight;
+    type OutboundXcmpMessageSource = XcmpQueue;
+    type XcmpMessageHandler = XcmpQueue;
+    type ReservedXcmpWeight = ReservedXcmpWeight;
+}
+
+impl parachain_info::Config for Test {}
+
+parameter_types! {
+    pub const DotLocation: MultiLocation = MultiLocation::X1(Parent);
+    pub const RelayNetwork: NetworkId = NetworkId::Kusama;
+    pub RelayChainOrigin: Origin = cumulus_pallet_xcm::Origin::Relay.into();
+    pub Ancestry: MultiLocation = Parachain(ParachainInfo::parachain_id().into()).into();
+}
+
+pub type LocationToAccountId = (
+    ParentIsDefault<AccountId>,
+    SiblingParachainConvertsVia<Sibling, AccountId>,
+    AccountId32Aliases<RelayNetwork, AccountId>,
+);
+
+pub type XcmOriginToCallOrigin = (
+    SovereignSignedViaLocation<LocationToAccountId, Origin>,
+    RelayChainAsNative<RelayChainOrigin, Origin>,
+    SiblingParachainAsNative<cumulus_pallet_xcm::Origin, Origin>,
+    SignedAccountId32AsNative<RelayNetwork, Origin>,
+    XcmPassthrough<Origin>,
+);
+
+parameter_types! {
+    pub const UnitWeightCost: Weight = 1;
+    pub DotPerSecond: (MultiLocation, u128) = (X1(Parent), 1);
+}
+
+// pub type LocalAssetTransactor =
+//     XcmCurrencyAdapter<ParaTokens::CurrencyAdapter, IsConcrete<DotLocation>, LocationToAccountId, AccountId, ()>;
+
+pub type LocalAssetTransactor = MultiCurrencyAdapter<
+    Tokens,
```

Comment:
is there a difference using `Currencies` here? or maybe they just impl the same trait

Reply:
yes, they impl the same trait